cairo-oracle-hint-processor/src/rpc_hint_processor.rs

```diff
@@ -109,74 +107,143 @@ impl<'a> Rpc1HintProcessor<'a> {
         );
         println!("let the oracle decide... Inputs: {data:?}");
 
-        let client = reqwest::blocking::Client::new();
+        let client = reqwest::blocking::ClientBuilder::new()
+            .timeout(Duration::from_secs(10)) // Short timeout for each request
+            .build()
+            .map_err(|e| {
+                HintError::CustomHint(Box::from(format!("Failed to create HTTP client: {}", e)))
+            })?;
 
-        let req = client.post(server_url.clone()).json(&data).send().expect(
-            format!("Couldn't connect to oracle server {server_url}. Is the server running?")
-                .as_str(),
-        );
+        // Polling parameters
+        let default_pooling_config = PoolingConfig {
+            max_attempts: 30,
+            polling_interval: 2,
+            timeout: 60,
+        };
 
-        let status_code = req.error_for_status_ref().map(|_| ());
-        let body = req.text().expect(
-            formatdoc! {
-                r#"
-                Response from oracle server can't be parsed as string."#
-            }
-            .as_str(),
-        );
+        let pooling_config = server_config
+            .pooling_config
+            .as_ref()
+            .unwrap_or(&default_pooling_config);
+
+        let max_attempts = pooling_config.max_attempts;
+        let polling_interval = Duration::from_secs(pooling_config.polling_interval);
+        let start_time = Instant::now();
+        let timeout = Duration::from_secs(pooling_config.timeout);
+
+        // Initial request to start the job
+        let response = client
+            .post(server_url.clone())
+            .json(&data)
+            .send()
+            .map_err(|e| {
+                HintError::CustomHint(Box::from(format!(
+                    "Failed to send request to oracle server {}: {}",
+                    server_url, e
+                )))
+            })?;
 
-        status_code.expect(
-            formatdoc! {
-                r#"
-                Received {body:?}.
-                Response status from oracle server not successful."#
-            }
-            .as_str(),
-        );
+        let response_body = response.text().map_err(|e| {
+            HintError::CustomHint(Box::from(format!("Failed to get response body: {}", e)))
+        })?;
+
+        println!("Initial response body: {}", response_body);
+
+        let response_json: serde_json::Value =
+            serde_json::from_str(&response_body).map_err(|e| {
+                HintError::CustomHint(Box::from(format!("Failed to parse response JSON: {}", e)))
+            })?;
+
+        let job_id = response_json

```

Comment:
It seems that now everything is a job on the other APIs, but for example for a `square_root` I think that this would be too much, maybe there could be an option for `long_request` that triggers the pooling if we know that its a long process

Reply:
You mean making the polling mechanism optional ? 

I can implement it for sure

Reply:
Yeah, think that maybe for a quick request for a square root or a matmul which would be done in Rust will be basically instantaneous, making the polling will make more complex that it would need to be, but if we know what its a long process que could just do `poll=True`

Reply:
Sounds good! Yes we should know in advance the what's a long request. I will implement `poll=true`